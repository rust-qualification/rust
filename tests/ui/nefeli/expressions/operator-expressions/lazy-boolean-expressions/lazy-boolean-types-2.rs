// Lazy Boolean Opearations: 8ee2a7bd-59c9-4479-9516-5519af52aa3b

pub fn main() {
    assert!(1 && 1); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types
    assert!(1.5 && 1.5); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types
    assert!(1.5 && 1); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types    
    assert!(false && 1); 
    //~^ ERROR mismatched types
    assert!('a' && 'b'); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types    
    assert!('a' || true); 
    //~^ ERROR mismatched types
    assert!("alpha" && "beta");
    //~^ ERROR mismatched types
    //~| ERROR mismatched types     
    assert!("alpha" || true); 
    //~^ ERROR mismatched types
}