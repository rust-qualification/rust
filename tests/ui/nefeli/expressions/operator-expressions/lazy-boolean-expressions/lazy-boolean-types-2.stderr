error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:8:13
   |
LL |     assert!(1 && 1); 
   |             ^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:8:18
   |
LL |     assert!(1 && 1); 
   |                  ^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:11:13
   |
LL |     assert!(1.5 && 1.5); 
   |             ^^^ expected `bool`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:11:20
   |
LL |     assert!(1.5 && 1.5); 
   |                    ^^^ expected `bool`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:14:13
   |
LL |     assert!(1.5 && 1); 
   |             ^^^ expected `bool`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:14:20
   |
LL |     assert!(1.5 && 1); 
   |                    ^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:17:22
   |
LL |     assert!(false && 1); 
   |             -----    ^ expected `bool`, found integer
   |             |
   |             expected because this is `bool`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:19:13
   |
LL |     assert!('a' && 'b'); 
   |             ^^^ expected `bool`, found `char`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:19:20
   |
LL |     assert!('a' && 'b'); 
   |                    ^^^ expected `bool`, found `char`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:22:13
   |
LL |     assert!('a' || true); 
   |             ^^^ expected `bool`, found `char`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:24:13
   |
LL |     assert!("alpha" && "beta");
   |             ^^^^^^^ expected `bool`, found `&str`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:24:24
   |
LL |     assert!("alpha" && "beta");
   |                        ^^^^^^ expected `bool`, found `&str`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types-2.rs:27:13
   |
LL |     assert!("alpha" || true); 
   |             ^^^^^^^ expected `bool`, found `&str`

error: aborting due to 13 previous errors

For more information about this error, try `rustc --explain E0308`.
